tests/test_electrolyzer.py

```diff
@@ -268,6 +319,133 @@ def test_electrolyzer_constant_efficiency(self):
         # Electrolyser
         efficiency = solution.parameters(0)["Electrolyzer_fc66.efficiency"]
         np.testing.assert_allclose(
-            results["Electrolyzer_fc66.Gas_mass_flow_out"] * efficiency,
+            results["Electrolyzer_fc66.Gas_mass_flow_out"] * efficiency * 3600,
             results["Electrolyzer_fc66.ElectricityIn.Power"],
         )
+        # Check input power values. Not really needed since the massflow check is equivalent
+        # np.testing.assert_allclose(
+        #     results["Electrolyzer_fc66.ElectricityIn.Power"],
+        #     [1.00000000e+08, 1.00000000e+08, 1.00000000e+08],
+        #     atol=1e-4,
+        # )
+        # Check output massflow values
+        np.testing.assert_allclose(
+            results["Electrolyzer_fc66.Gas_mass_flow_out"],
+            [440.91710758, 440.91710758, 440.91710758],
+            atol=1e-4,
+        )
+
+    def test_electrolyzer_equality_constraint(self):
+        """
+        This test is to check the functioning the example with an offshore wind farm in combination
+        with an electrolyzer and hydrogen storage. The electrolyzer is modelled as the option
+        LINEARIZED_THREE_LINES_EQUALITY.
+
+        Checks:
+        - Check that only one line is activated
+        - Check that the expected lines are activated, depending on the input power
+        - Check that the output massflow lies on the line segment
+        - Check hardcoded values
+
+        """
+        import models.unit_cases_electricity.electrolyzer.src.example as example
+        from models.unit_cases_electricity.electrolyzer.src.example import (
+            MILPProblemEquality,
+        )
+
+        base_folder = Path(example.__file__).resolve().parent.parent
+
+        solution = run_esdl_mesido_optimization(
+            MILPProblemEquality,
+            base_folder=base_folder,
+            esdl_file_name="h2.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="timeseries_equality_constraints.csv",
+        )
+
+        results = solution.extract_results()
+
+        # Check that there is only one activated line per timestep
+        for timestep in range(len(results["Electrolyzer_fc66__line_0_active"])):
+            np.testing.assert_allclose(
+                (
+                    results["Electrolyzer_fc66__line_0_active"][timestep]
+                    + results["Electrolyzer_fc66__line_1_active"][timestep]
+                    + results["Electrolyzer_fc66__line_2_active"][timestep]
+                ),
+                1.0,
+            )
```

Comment:
shouldn't this check be assert_array_less, because this sum should also be possible to be 0.
Ideally I would see the check including the asset_is_swithced_on variable.

Reply:
I didn't include that as the test is tailored to the csv, and not including asset_switched_on makes this test most restrictive. I changed it anyways